src/main/java/com/cx/restclient/CxSASTClient.java

```diff
@@ -641,8 +696,9 @@ private long createScan(long projectId) throws IOException {
     }
 
 
-    private CxID createRemoteSourceRequest(long projectId, HttpEntity entity, String sourceType, boolean isSSH) throws IOException {
-        return httpClient.postRequest(String.format(SAST_CREATE_REMOTE_SOURCE_SCAN, projectId, sourceType, isSSH ? "ssh" : ""), isSSH ? null : CONTENT_TYPE_APPLICATION_JSON_V1,
+    private CxID createRemoteSourceRequest(long projectId, String apiVersion,HttpEntity entity, String sourceType, boolean isSSH) throws IOException {
+    	
+        return httpClient.postRequest(String.format(SAST_CREATE_REMOTE_SOURCE_SCAN, projectId, sourceType, isSSH ? "ssh" : ""), isSSH ? null : CONTENT_TYPE_APPLICATION_JSON_V1, apiVersion,
```

Comment:
The function should have been called for the API SAST_CREATE_REMOTE_SOURCE_SCAN,instead of SCAN_WITH_SETTINGS_URL